cmd/stolonctl/status.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright (C) 2018 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"text/tabwriter"
+	"time"
+
+	"github.com/gravitational/stolon-app/internal/stolonctl/pkg/cluster"
+
+	"github.com/gravitational/trace"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var statusCmd = &cobra.Command{
+	Use:   "status",
+	Short: "Show status of PostgreSQL cluster",
+	RunE:  status,
+}
+
+func init() {
+	stolonctlCmd.AddCommand(statusCmd)
+}
+
+func status(ccmd *cobra.Command, args []string) error {
+	clusterStatus, err := Status()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	PrintStatus(clusterStatus)
+	return nil
+}
+
+// Status returns status of Stolon cluster(Pods state and ClusterView)
+func Status() (*cluster.Status, error) {
+	clusterConfig := &cluster.Config{
+		KubeConfig:         kubeConfig,
+		Name:               clusterName,
+		Namespace:          namespace,
+		KeepersPodFilter:   keepersFilter,
+		SentinelsPodFilter: sentinelsFilter,
+		EtcdEndpoints:      etcdEndpoints,
+		EtcdCertFile:       etcdCertFile,
+		EtcdKeyFile:        etcdKeyFile,
+		EtcdCAFile:         etcdCAFile,
+	}
+
+	status, err := cluster.GetStatus(clusterConfig)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return status, nil
+}
+
+// PrintStatus prints status of Stolon cluster to stdout
+func PrintStatus(status *cluster.Status) {
+	w := new(tabwriter.Writer)
+
+	var (
+		minwidth int
+		tabwidth = 8
+		padding  = 2
+		flags    uint
+		padchar  byte = '\t'
+	)
+	w.Init(os.Stdout, minwidth, tabwidth, padding, padchar, flags)
+	fmt.Fprintln(w, "NAME\tPOD_STATUS\tREADY\tIP\tNODE\tAGE\tKEEPER_ID\tKEEPER_HEALTHY\tMASTER")
```

Comment:
At some point, it might be worth using something like https://github.com/olekukonko/tablewriter that can automatically do the table generation and keep everything aligned.